domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt

```diff
@@ -730,7 +730,13 @@ class ExplorationProgressController @Inject constructor(
         // Follow the answer's outcome to another part of the graph if it's different.
         val ephemeralState = computeBaseCurrentEphemeralState()
         when {
-          answerOutcome.destinationCase == AnswerOutcome.DestinationCase.STATE_NAME -> {
+          answerOutcome.destinationCase == AnswerOutcome.DestinationCase.STATE_NAME &&
+            (
+              !enableFlashbackSupport.value ||
+                answerOutcome.labelledAsCorrectAnswer ||
+                !explorationProgress.stateDeck.isFlashbackViewed()
```

Comment:
Is this possibly a redudant condition?
```suggestion
                !explorationProgress.stateDeck.isFlashbackViewed()
```

Reply:
This condition ensures that the flashback button is not shown more than once after it has already been viewed. In this case, we fall into the second condition, which handles the pending state.